pandas/tests/sparse/test_series.py

```diff
@@ -529,19 +529,17 @@ def _compare(idx):
         tm.assert_series_equal(sp.take([0, 1, 2, 3, 4]), exp)
 
         with tm.assert_produces_warning(FutureWarning):
+            sp.take([1, 5], convert=True)
+
+        with tm.assert_produces_warning(FutureWarning):
             sp.take([1, 5], convert=False)
 
     def test_numpy_take(self):
         sp = SparseSeries([1.0, 2.0, 3.0])
         indices = [1, 2]
 
-        # gh-17352: older versions of numpy don't properly
-        # pass in arguments to downstream .take() implementations.
-        warning = FutureWarning if _np_version_under1p12 else None
```

Comment:
Is there a reason this is not the case anymore?

Reply:
Actually, I need to put that back.  However, I different default will be needed, as `convert=None` breaks downstream compatibility with `numpy`:

https://travis-ci.org/pandas-dev/pandas/jobs/285909832

Reply:
What is the problem with None for numpy? You linked to a sparse failure, but in the sparse code you left the default as None.

If it is needed, another solution might be to push it into the kwargs? (so the -1 is not visible)

BTW, can you push new commits instead of squashing into the previous one? That makes it easier to say what you changed.

Reply:
Oh, sorry, missed your message before I pushed.

The problem is that `numpy`'s `take` implementation for slightly older versions passes in `None` for one of its parameters *by position*.  This parameter corresponds to the `convert` parameter for our version of `take`.  It's messy compatibility stuff that I had to wade through at some point that I patched on both sides.